File: sdkwork-ai-sdk-swift/Sources/API/Videos2Api.swift
import Foundation

public class Videos2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// List videos
    public func listVideos(params: [String: Any]? = nil) async throws -> OpenAiVideoList? {
        let response = try await client.get(ApiPaths.aiPath("/v1/videos"), params: params)
        return response as? OpenAiVideoList
    }

    /// Create video
    public func createVideo(body: VideoGenerationRequest) async throws -> OpenAiVideo? {
        let response = try await client.post(ApiPaths.aiPath("/v1/videos"), body: body)
        return response as? OpenAiVideo
    }

    /// Retrieve video
    public func retrieveVideo(video_id: String) async throws -> OpenAiVideo? {
        let response = try await client.get(ApiPaths.aiPath("/v1/videos/\(video_id)"))
        return response as? OpenAiVideo
    }

    /// Delete video
    public func deleteVideo(video_id: String) async throws -> OpenAiVideoDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/v1/videos/\(video_id)"))
        return response as? OpenAiVideoDeleteResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/Videos2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Videos2Api
    {
        private readonly HttpClient _client;

        public Videos2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// List videos
        /// </summary>
        public async Task<OpenAiVideoList?> ListVideosAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<OpenAiVideoList>(ApiPaths.AiPath("/v1/videos"), query);
        }

        /// <summary>
        /// Create video
        /// </summary>
        public async Task<OpenAiVideo?> CreateVideoAsync(VideoGenerationRequest body)
        {
            return await _client.PostAsync<OpenAiVideo>(ApiPaths.AiPath("/v1/videos"), body);
        }

        /// <summary>
        /// Retrieve video
        /// </summary>
        public async Task<OpenAiVideo?> RetrieveVideoAsync(string video_id)
        {
            return await _client.GetAsync<OpenAiVideo>(ApiPaths.AiPath($"/v1/videos/{video_id}"));
        }

        /// <summary>
        /// Delete video
        /// </summary>
        public async Task<OpenAiVideoDeleteResponse?> DeleteVideoAsync(string video_id)
        {
            return await _client.DeleteAsync<OpenAiVideoDeleteResponse>(ApiPaths.AiPath($"/v1/videos/{video_id}"));
        }
    }
}
